cdap-proto/src/main/java/co/cask/cdap/proto/ProtoTrigger.java

```diff
@@ -35,10 +37,37 @@
    * Represents all known trigger types in REST requests/responses.
    */
   public enum Type {
-    TIME,
-    PARTITION,
-    STREAM_SIZE,
-    PROGRAM_STATUS
+    TIME("time"),
+    PARTITION("partition"),
+    STREAM_SIZE("stream-size"),
+    PROGRAM_STATUS("program-status");
+
+    private static final Map<String, Type> CATEGORY_MAP;
+
+    static {
+      CATEGORY_MAP = new HashMap<>();
+      for (Type type : Type.values()) {
+        CATEGORY_MAP.put(type.getCategoryName(), type);
+      }
+    }
+
+    private final String cateogryName;
+
+    Type(String cateogryName) {
+      this.cateogryName = cateogryName;
+    }
+
+    public String getCategoryName() {
+      return cateogryName;
+    }
+
+    public static Type valueOfCategoryName(String categoryName) {
```

Comment:
this will break the compatibility of pre-4.3 JSON object and 4.3 deserialization